File: ruby/tests/reverse/reverse_test.rb
# frozen_string_literal: true

require 'test_helper'
require 'reverse/reverse'

describe Reverse do
  def setup
    @tests = [
      { word: 'cat', want: 'tac' },
      { word: 'alphabet', want: 'tebahpla' },
      { word: '日本語', want: '語本日' }
    ]
  end

  describe '.iterate' do
    it 'reverses a word' do
      @tests.each do |tc|
        _(Reverse.iterate(tc[:word])).must_equal(tc[:want])
      end
    end
  end

  describe '.recursive' do
    it 'reverses a word' do
      @tests.each do |tc|
        _(Reverse.recursive(tc[:word])).must_equal(tc[:want])
      end
    end
  end

  describe '.std' do
    it 'reverses a word' do
      @tests.each do |tc|
        _(Reverse.std(tc[:word])).must_equal(tc[:want])
      end
    end
  end
end

File: go/tests/reverse/reverse_test.go
package reverse_test

import (
	"fmt"
	"github/ricardoalmeida/polyglot/go/src/reverse"
	"testing"
)

func TestIterate(t *testing.T) {
	for _, tc := range tests {
		t.Run(fmt.Sprintf("%v", tc.word), func(t *testing.T) {
			got := reverse.Iterate(tc.word)
			if got != tc.want {
				t.Fatalf("Iterate() = %v; want %v", got, tc.want)
			}
		})
	}
}

var tests = []struct {
	word string
	want string
}{
	{"cat", "tac"},
	{"alphabet", "tebahpla"},
	{"日本語", "語本日"},
}
